### ISSUE TITLE
Weather not working due to Yahoo API changes

### ISSUE THREAD

**Issue by jest** [quodroc/HumbleNewTabPage#65]
According to https://developer.yahoo.com/weather/:

> As of Thursday, Jan. 3, 2019, the weather.yahooapis.com and query.yahooapis.com for Yahoo Weather API will be retired. 

My DNS no longer resolve query.yahooapis.com anymore. Probably change to API usage must be implemented (including OAuth authentication?)

**Comment by mike2nl** [comment]
Same here on 3 PC's.
It started some hours ago.

**Comment by mike2nl** [comment]
Message on Yahoo site ir red on top:

Important EOL Notice: As of Thursday, Jan. 3, 2019, the weather.yahooapis.com and query.yahooapis.com for Yahoo Weather API will be retired. 
To continue using our free Yahoo Weather APIs, use https://weather-ydn-yql.media.yahoo.com/forecastrss. Contact yahoo-weather-ydn-api@oath.com for credentials to onboard to this free Yahoo Weather API service.

**Comment by quodroc** [comment]
It's not as simple as changing the URL. Even if I contact them for credentials, there are a few issues:
- I need some way to look up the location id (woeid) from the city/country text the user types in (currently done using the now defunct YQL api)
- The forecastrss response is XML instead of JSON (not too big a deal)
- Even though the credentials are free, I would have no way of preventing anyone from stealing them and potentially getting my account banned

I'm considering either removing the weather feature entirely (it's been nothing but trouble over the years, and not even very accurate), or switching to OpenWeatherMap which has a much nicer API (though securing the credentials would still be a problem)

**Comment by jest** [comment]
> I'm considering either removing the weather feature entirely (it's been nothing but trouble over the years, and not even very accurate), or switching to OpenWeatherMap which has a much nicer API.

OWM also requires API key (https://openweathermap.org/faq#error401). I don't know how important it is to protect the key, but you can't do much as long as it is being visible in URL sent from the extension (e.g in "developer tools/network"). Yet, according to their FAQ:

> For FOSS developers: we welcome free and open source software and are willing to help you. If you want to use OWM data in your free software application please register an API key and file a ticket describing your application and API key registered. OWM will review your request lift access limits for your key if used in open source application.

**Comment by rpitera** [comment]
OWM is available in JSON.  Also, could you not have the user request their own API key?  It's free and easy and would eliminate any limits on a single key.  Also OWM can do locations in various forms including ZIP codes.

**Comment by atwright147** [comment]
Probably worth removing it until a decision is made, or at least putting some sort of notification to explain what is going on somewhere

**Comment by rpitera** [comment]
> Probably worth removing it until a decision is made, or at least putting some sort of notification to explain what is going on somewhere

I agree; there's a couple of components that could use some kind of a tag like that.  USPS and FedEx sensors haven't worked in quite some time yet are still listed in the components section.

**Comment by 4doge** [comment]
@quodroc can we implement https://www.metaweather.com/api/ instead of Yahoo?

**Comment by quodroc** [comment]
I have disabled the weather feature for now. Will work on OpenWeatherMap support in the future. Metaweather looks interesting as well.
